ex2_services/Services/source/payload/iris/iris_service.c

```diff
@@ -177,12 +177,20 @@ SAT_returnState iris_service_app(csp_packet_t *packet) {
          * 2. Transfer image
          */
         uint32_t image_length;
+        uint16_t image_count;
+        char filename[100];
 
-        status = iris_get_image_length(&image_length);
-        IRIS_SERVICE_IMAGE_TRANSFER_DELAY; // 100 ms delay
+        status = iris_get_image_count(&image_count);
+        if (status == IRIS_HAL_OK) {
+            for (int i = 0; i < image_count; i++) {
+                status = iris_get_image_length(&image_length);
+                IRIS_SERVICE_IMAGE_TRANSFER_DELAY; // 100 ms delay
```

Comment:
This comment has lots of potential to be wrong. Suggest removing it, readers can simply look at the definition